src/main/java/org/broadinstitute/hellbender/tools/spark/sv/evidence/EvidenceTargetLink.java

```diff
@@ -0,0 +1,121 @@
+package org.broadinstitute.hellbender.tools.spark.sv.evidence;
+
+import com.esotericsoftware.kryo.DefaultSerializer;
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import org.broadinstitute.hellbender.tools.spark.sv.utils.PairedStrandedIntervals;
+import org.broadinstitute.hellbender.tools.spark.sv.utils.SVInterval;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@DefaultSerializer(EvidenceTargetLink.Serializer.class)
+public class EvidenceTargetLink {
+    private static final SVInterval.Serializer intervalSerializer = new SVInterval.Serializer();
+
+    final SVInterval source;
+    final boolean sourceForwardStrand;
+    final SVInterval target;
+    final boolean targetForwardStrand;
+    final int splitReads;
+    final int readPairs;
+
+    public EvidenceTargetLink(final SVInterval source, final boolean sourceForwardStrand, final SVInterval target, final boolean targetForwardStrand, final int splitReads, final int readPairs) {
+        this.source = source;
+        this.sourceForwardStrand = sourceForwardStrand;
+        this.target = target;
+        this.targetForwardStrand = targetForwardStrand;
+        this.splitReads = splitReads;
+        this.readPairs = readPairs;
+    }
+
+    public EvidenceTargetLink(final Kryo kryo, final Input input) {
+        this.source = intervalSerializer.read(kryo, input, SVInterval.class);
+        this.sourceForwardStrand = input.readBoolean();
+        this.target = intervalSerializer.read(kryo, input, SVInterval.class);
+        this.targetForwardStrand = input.readBoolean();
+
+        this.splitReads = input.readInt();
+        this.readPairs = input.readInt();
+    }
+
+    protected void serialize(final Kryo kryo, final Output output ) {
+        intervalSerializer.write(kryo, output, source);
+        output.writeBoolean(sourceForwardStrand);
+        intervalSerializer.write(kryo, output, target);
+        output.writeBoolean(targetForwardStrand);
+
+        output.writeInt(splitReads);
+        output.writeInt(readPairs);
+    }
+
+    public String toBedpeString(ReadMetadata readMetadata) {
+        final Map<Integer, String> contigIdToContigNameMap =
```

Comment:
You don't need this.  ReadMetadata stores the inverse map.  Use readMetadata.getContigName( int id ) to turn a contigID into a contigName.